### PR TITLE
Fix #172 by adding parsing step to save_metadata

### LINKED ISSUE #172: Not using chapter name from audible-cli when using m4b without --chaptered
When using `./AAXtoMP3 -e:m4b --use-audible-cli-data -A foo bar.aax` e. g. without `-c` or `--chaptered` the data is still taken from chapters.txt, which doesn't contain the correct chapter naming. Therefore the resulting single m4b doesn't have the correct chapter titles.

I guess the script would need to parse the json file and write the correct chapter names into the created chapters.txt in case audibleCli=1, before applying the chapters.txt to the single m4b file.

### PR DESCRIPTION
**Pull request by suntorytimed**
* creates a correctly formatted chapter.txt file
  with titles from audible-cli .json
* uses this chapter.txt instead of ffprobe output
  in case audible-cli data should be used

### CODE REVIEW

File: AAXtoMP3 [hunk 1]
```diff
@@ -381,6 +381,9 @@ trap 'rm -r -f "${working_directory}"' EXIT
 # Set up some basic working files ASAP.  Note the trap will clean this up no matter what.
 working_directory=`mktemp -d 2>/dev/null || mktemp -d -t 'mytmpdir'`
 metadata_file="${working_directory}/metadata.txt"
```
Comment (suntorytimed) [inline AAXtoMP3:383]: Should I add a comment for this line?

File: AAXtoMP3 [hunk 2]
```diff
@@ -381,6 +381,9 @@ trap 'rm -r -f "${working_directory}"' EXIT
...
+if [[ "${audibleCli}" == "1" && "${mode}" == "single" ]] ; then
```
Comment (suntorytimed) [inline AAXtoMP3:384]: Should I add a comment for this line?

Comment (fabh2o) [reply AAXtoMP3:384]: Yes, I think it would be nice to shortly explain why we create a new temp file.

Comment (suntorytimed) [reply AAXtoMP3:384]: Okay, done.

File: AAXtoMP3 [hunk 3]
```diff
@@ -510,6 +517,18 @@ save_metadata() {
...
+    # In case we want to use a single file m4b we need to extract the
+    # chapter titles from the .json generated by audible–cli and store
+    # them correctly formatted for mp4chaps in a chapter.txt
+    if [ "${mode}" == "single" ]; then
+      jq -r \
+        'def pad(n): tostring | if (n > length) then ((n - length) * "0") + . else . end;
+        .content_metadata.chapter_info.chapters |
+        to_entries |
+        .[] |
+        "CHAPTER\((.key | pad(2)))=\(((.value.start_offset_ms / (1000*60*60)) %24 | floor | pad(2))):\(((.value.start_offset_ms / (1000*60)) %60 | floor | pad(2))):\(((.value.start_offset_ms / 1000) %60 | floor | pad(2))).\((.value.start_offset_ms % 1000 | pad(3)))
```
Comment (fabh2o) [inline AAXtoMP3:529]: The padding for the chapter number is not necessary
```suggestion
        "CHAPTER\((.key))=\(((.value.start_offset_ms / (1000*60*60)) %24 | floor | pad(2))):\(((.value.start_offset_ms / (1000*60)) %60 | floor | pad(2))):\(((.value.start_offset_ms / 1000) %60 | floor | pad(2))).\((.value.start_offset_ms % 1000 | pad(3)))
```

File: AAXtoMP3 [hunk 4]
```diff
@@ -510,6 +517,18 @@ save_metadata() {
...
+CHAPTER\((.key | pad(2)))NAME=\(.value.title)"' "${extra_chapter_file}" > "${tmp_chapter_file}"
```
Comment (fabh2o) [inline AAXtoMP3:530]: The padding for the chapter number is not necessary
```suggestion
CHAPTER\((.key))NAME=\(.value.title)"' "${extra_chapter_file}" > "${tmp_chapter_file}"
```

File: AAXtoMP3 [hunk 5]
```diff
@@ -882,8 +901,14 @@ do
...
     # ----
     # ffmpeg seems to copy only chapter position, not chapter names.
+    # use already created chapter.txt from save_metadata() in case
+    # audible-cli data is used else use ffprobe to extract from m4b
     if [[ ${container} == "mp4" && $(type -P mp4chaps) ]]; then
-      ffprobe -i "${aax_file}" -print_format csv -show_chapters 2>/dev/null | awk -F "," '{printf "CHAPTER%02d=%02d:%02d:%02.3f\nCHAPTER%02dNAME=%s\n", NR, $5/60/60, $5/60%60, $5%60, NR, $8}' > "${output_directory}/${currentFileNameScheme}.chapters.txt"
+      if [ "${audibleCli}" == "1" ]; then
+        mv "${tmp_chapter_file}" "${output_directory}/${currentFileNameScheme}.chapters.txt"
+      else
+        ffprobe -i "${aax_file}" -print_format csv -show_chapters 2>/dev/null | awk -F "," '{printf "CHAPTER%02d=%02d:%02d:%02.3f\nCHAPTER%02dNAME=%s\n", NR, $5/60/60, $5/60%60, $5%60, NR, $8}' > "${output_directory}/${currentFileNameScheme}.chapters.txt"
```
Comment (fabh2o) [inline AAXtoMP3:910]: The padding for the chapter number is not necessary
```suggestion
        ffprobe -i "${aax_file}" -print_format csv -show_chapters 2>/dev/null | awk -F "," '{printf "CHAPTER%d=%02d:%02d:%02.3f\nCHAPTER%dNAME=%s\n", NR, $5/60/60, $5/60%60, $5%60, NR, $8}' > "${output_directory}/${currentFileNameScheme}.chapters.txt"
```

File: AAXtoMP3 (hunk 3 above)
Comment (suntorytimed) [reply AAXtoMP3:529]: ah, good to know. Even better :)

**Comment by suntorytimed** [comment]
@fabh2o tested again with the changes from your review applied and it works as expected.

**Comment by KrumpetPirate** [comment]
Thanks for the contribution!

**Comment by suntorytimed** [comment]
Thanks for merging and thanks to @fabh2o for his original idea with jq that helped a lot!

**Pull request merged**